### DISCUSSION TITLE
Dealing with ionisation

### DISCUSSION THREAD

**Discussion by PhysicsDan** [epochpic/epoch#342] [Q&A] [answered]
I have the following species definitions - where I am defining the number density of neutral and single ionised carbon ions, which can be further ionised within the EPOCH simulation.

```
begin:species
  name = carbon
  charge = 0
  mass = 12*amu
  npart_per_cell = npart_default

  ionisation_energies = (11.26*ev,24.38*ev,47.89*ev,64.49*ev,392.1*ev,490.0*ev)
  ionisation_electron_species = electron

  number_density = 'epoch_files_cart/density_neutrals_8000_5000.dat'
  temp = 'epoch_files_cart/temperature_neutrals_8000_5000.dat'

end:species

begin:species
  name = carbon1
  charge = 1
  mass = 12*amu
  npart_per_cell = npart_default

  ionisation_energies = (24.38*ev,47.89*ev,64.49*ev,392.1*ev,490.0*ev)
  ionisation_electron_species = electron

  number_density = 'epoch_files_cart/density_ions_8000_5000.dat'
  temp = 'epoch_files_cart/temperature_ions_8000_5000.dat'

end:species
```

Unfortunately I get this error - which I believe means that when I define `carbon1` in the input deck and EPOCH creates a `carbon1` species when considering the ionisation of the `carbon` species, one of the two gets overwritten.

```
 Specify output directory
 Name of species 1 is carbon
 Name of species 2 is carbon1
 Name of species 3 is carbon2
 Name of species 4 is carbon3
 Name of species 5 is carbon4
 Name of species 6 is carbon5
 Name of species 7 is carbon6
 Name of species 8 is carbon61
 Name of species 9 is carbon62
 Name of species 10 is carbon63
 Name of species 11 is carbon64
 Name of species 12 is carbon65
 Name of species 13 is electron
 *** WARNING ***
 Incorrect number of release species specified for carbon. Using only first specified.
```

**Comment by PhysicsDan** [comment, marked as answer]
Just found the solution.

I removed the lines
```
  ionisation_energies = (24.38*ev,47.89*ev,64.49*ev,392.1*ev,490.0*ev)
  ionisation_electron_species = electron
```
From my `carbon1` species declaration. I suppose that was resulting in redefining the subsequent ionisation levels.

I'll leave this here in case others make a similar mistake.

**Comment by DanRRRR** [comment]
I finally decided to look at new coll.ionization feature in 4.19 everyone were waiting for and sorry got tons of questions. Here are the major ones

1) Can anyone point me on Help with description of coll.ionization part done in latest 4.19 ?

2) Do i correctly understand that with this new ionization included you do not need to define electrons fraction as it is now can be defined by the ionization process in initial settings? Manual settings of fractions of electrons and atoms actually now should go into conflict with coll.ionization ON (???)

3) Can the charge of each individual particle now be included in the Particles output together with coordinates, momentum and weight? Which specific variable is it? Is it current%charge (and current => species_list(ispecies)%attached_list%head) ? 

4) Were simple but complete examples of input deck showing how to use new coll ionization feature prepared? I see one example or the one that looks like one but coll.ionization there is initially commented out because of warning that the ionization "slows down the simulation". OK, of course we accept it slows down, but if i un-comment these lines do i need to comment some older others? This is unclear given Q#2 above.

5) and by the way how much coll.ionization slows down the simulations and what causes it? The LOG or  EXP functions which are typically present in appropriate crossections (Lotz for example) or just more equations now included?

**Reply by Status-Mirror** [reply]
1. The physics behind the collisional ionisation model is explained here: https://aip.scitation.org/doi/abs/10.1063/5.0126336, while documentaton on collisional ionisation is given in the ionisation section here: https://epochpic.github.io/documentation/input_deck/input_deck_species.html
2. When you say fraction, I'm guessing you mean the `frac` key in the species block. This controls how many macro-particles are **initially** assigned to each species. The automatically generated species initially start empty, so `frac` has no effect on them. These species will be populated when ionisation occurs.
3. It is `current%charge` if you compile the code with `-DPER_PARTICLE_CHARGE_MASS`, otherwise the charge is present in `species_list(ispecies)%charge`. You can output this by adding `charge=always` to your output block: https://epochpic.github.io/documentation/input_deck/input_deck_output_block.html
4. Yes, you only need to uncomment the collisions block to switch it on in that deck. The `ionise=T` will automatically generate separate ion species. These will initially be empty - macro-particles will be added to these species after ionisation events.
5. More equations, and also we need to sort particles to their nearest cells so we only collide spatially local particles together. See the Morris paper in 1. for a full description of how this model was implemented.

Cheers,
Stuart

**Comment by DanRRRR** [comment]
Thanks, Stuart
